x-pack/filebeat/input/gcppubsub/config.go

```diff
@@ -72,7 +72,13 @@ func defaultConfig() config {
 	c.ForwarderConfig = harvester.ForwarderConfig{
 		Type: "gcp-pubsub",
 	}
+	// Since we now support multiple beat.Clients, having multiple NumGoroutines
+	// greatly improves ingestion performance. Hence, default values are being
+	// adjusted to get the most of the input.
+	// It is not increased too high to cause high CPU usage.
 	c.Subscription.NumGoroutines = 2
+	// The input gets blocked until flush.min_events or flush.timeout is reached.
+	// Hence max_outstanding_message has to be atleast flush.min_events to avoid this blockage.
 	c.Subscription.MaxOutstandingMessages = 1600
```

Comment:
It's unfortunate that there is not a globally visible variable/constant that could be used for this that would ensure that this is both explained in code an robust to source mutation.

Is there a way for the input to know `flush.min_events` and adjust this to be `max(flush.min_events, max_outstanding_messages)`?

Reply:
Since the queue config is defined inside publisher, I don't see any quick way to get this check without a considerable refactor of Input.

Reply:
Indeed. It was a wistful desire rather than an expectation of change.